src/components/RegisterForm.jsx

```diff
@@ -0,0 +1,109 @@
+import Title from "./Title";
+import Form from "./Form";
+import Input from "./Input";
+import Button from "./Button";
+import {useForm} from "../hooks";
+import {useState} from "react";
+import until from "../utils/until";
+import AuthService from "../services/AuthService";
+import {handleLogin, isLoggedIn} from "../utils/auth";
+import {Redirect} from "react-router-dom";
+
+const RegisterForm = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  const register = async () => {
+    setIsLoading(true);
+    setError(null);
+
+    const user = {
+      firstName: values.firstName,
+      lastName: values.lastName,
+      email: values.email,
+      password: values.password,
+      language: navigator.language.slice(0, 2)
+    };
+
+    const [err, result] = await until(AuthService.register(user));
+
+    if (err) {
+      setError({message: err.message});
+      setIsLoading(false);
+    }
+    if (result && result.data.errors) {
+      setError({message: "Invalid credentials provided"});
+      setIsLoading(false);
+    }
+
+    const {data = {}} = result.data || {};
+
+    if (data && data.user && data.token) {
+      handleLogin({
+        ...data.user,
+        token: data.token
+      });
+      setIsLoading(false);
+    }
+  };
+
+  const {values, handleChange, handleSubmit} = useForm(register, {
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+    language: navigator.language.slice(0, 2)
```

Comment:
This one doesn't have to be filled, you can simply keep it empty
```suggestion
    language: ""
```